cpp/oneapi/dal/algo/kmeans_init/backend/cpu/compute_kernel_dense.cpp

```diff
@@ -57,37 +69,72 @@ static compute_result call_daal_kernel(const context_cpu& ctx,
     const size_t len_output                                 = 1;
     daal::data_management::NumericTable* output[len_output] = { daal_centroids.get() };
 
-    interop::status_to_exception(
-        interop::call_daal_kernel<Float, daal_kmeans_init_dense_kernel_t>(ctx,
-                                                                          len_input,
-                                                                          input,
-                                                                          len_output,
-                                                                          output,
-                                                                          &par,
-                                                                          *(par.engine)));
+    if constexpr (std::is_same_v<Method, method::dense>)
```

Comment:
I think that checks can be done earlier in `check_preconditions()`. Instead of many if's you can use `statis_assert` with `detail::is_one_of`: https://github.com/oneapi-src/oneDAL/blob/master/cpp/oneapi/dal/detail/common.hpp#L25
For example, `static_assert(detail::is_one_of<Method, method::dense, method::random_dense, method::plus_plus_dense>::value)`

Reply:
You don't have to adress the comment above, but I think you should not copy-past all parameters of `call_daal_kernel` for every `Method` since they are the same. You can create a struct that will convert from new method tag to old one:

```cpp
template <typename Method>
struct convert_method { using type = void };

template <> struct convert_method<method::dense> { using type = daal_kmeans_init::defaultDense };
template <> struct convert_method<method::random_dense> { using type = daal_kmeans_init::randomDense};
```

And then use it:
```cpp
using daal_method_t = typename convert_method<Method>::type;
static_assert(!is_same_v<daal_method_t, void>, "illegal method");
interop::call_daal_kernel<Float, daal_method_t>(ctx, ... );
```

Reply:
done